packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -246,12 +259,17 @@ class BuildInfo {
     };
   }
 
+
   /// Convert to a structured string encoded structure appropriate for usage as
   /// environment variables or to embed in other scripts.
   ///
   /// Fields that are `null` are excluded from this configuration.
   Map<String, String> toEnvironmentConfig() {
-    return <String, String>{
+    final Map<String, String> map = <String, String>{};
+    defineConfigJsonMap.forEach((String key, dynamic value) {
```

Comment:
```suggestion
    defineConfigJsonMap.forEach((String key, Object? value) {
```

Reply:
prefer `Object?` over `dynamic`, as it preserves compiler type checks that `dynamic` will override